types/chai-as-promised/index.d.ts

```diff
@@ -52,6 +52,7 @@ declare global {
             Arguments: PromisedAssertion;
             equal: PromisedEqual;
             equals: PromisedEqual;
+            containSubset: PromisedContainSubset;
```

Comment:
How does this work? I don't see this in their repo: https://github.com/search?q=repo%3Achaijs%2Fchai-as-promised%20containSubset&type=code

Reply:
that works because `chai-as-promised` will iterate over all the available assertions dynamically and install new assertions based on that. It doesn't have a list of assertions that it maintains manually. This way, it will also work with assertions that come from plugins, for example. I have been using `chai-as-promised` with `chai-subset` for years for example

Reply:
Are any of the other declarations here for addon packages?

It seems strange to declare that every addon exists, even if they aren't installed, or aren't supported (like for older versions of chai).

Reply:
`containSubset` is not an add-on anymore, it was [merged](https://github.com/chaijs/chai/pull/1664) into chai proper, therefore adding it officially here now. What I was saying is that in the past I was using it as an add-on, and in that case I manually added the typings in my individual projects

Reply:
So, yes, but I'm specifically talking about people using `chai<5.2.0` with `chai-subset` and `chai-as-promised`.

Reply:
those people get the typing for free now?
not sure I get what's the concern...

Reply:
Sorry, misspoke; I meant `chai<5.2.0` and `chai-as-promise` but _without_ `chai-subset`, so they'd see a declaration which wouldn't work because nothing provides it.

It's probably unavoidable; it'd be nicer if the things tacked onto chai for promises were derived from the rest of chai, but I suspect that wouldn't actually work due to circularities without a major restructuring.